src/nbl/video/CVulkanImage.h

```diff
@@ -5,48 +5,37 @@
 #ifndef __NBL_VIDEO_C_VULKAN_IMAGE_H_INCLUDED__
 #define __NBL_VIDEO_C_VULKAN_IMAGE_H_INCLUDED__
 
-#include <volk.h>
 #include "nbl/video/IGPUImage.h"
-#include "nbl/video/CVKLogicalDevice.h"
 
-namespace nbl
-{
-namespace video
-{
+#define VK_NO_PROTOTYPES
+#include "vulkan/vulkan.h"
 
-class CVulkanImage final : public IGPUImage, public IDriverMemoryAllocation
+namespace nbl::video
 {
-	protected:
-		virtual ~CVulkanImage();
 
-	private:
-		CVKLogicalDevice* m_vkdevice;
-		VkImage m_vkimg;
+class ILogicalDevice;
 
+class CVulkanImage : public IGPUImage
+{
 	public:
-		//! constructor
-		CVulkanImage(CVKLogicalDevice* _vkdev, IGPUImage::SCreationParams&& _params);
-		CVulkanImage(CVKLogicalDevice* _vkdev, IGPUImage::SCreationParams&& _params, VkImage _vkimg) :
-			IGPUImage(_vkdev, std::move(_params)), m_vkdevice(_vkdev), m_vkimg(_vkimg)
-		{
+		// CVulkanImage(ILogicalDevice* _vkdev, IGPUImage::SCreationParams&& _params);
+		CVulkanImage(core::smart_refctd_ptr<ILogicalDevice>&& _vkdev, IGPUImage::SCreationParams&& _params, VkImage _vkimg) :
+			IGPUImage(std::move(_vkdev), std::move(_params)), m_vkImage(_vkimg)
+		{}
 
-		}
+		inline VkImage getInternalObject() const { return m_vkImage; }
 
-		inline VkImage getInternalObject() const { return m_vkimg; }
-
-		// TODO below
-		inline size_t getAllocationSize() const override { return this->getImageDataSizeInBytes(); }
-		inline IDriverMemoryAllocation* getBoundMemory() override { return this; }
-		inline const IDriverMemoryAllocation* getBoundMemory() const override { return this; }
+		// Todo(achal)
+		inline IDriverMemoryAllocation* getBoundMemory() override { return nullptr; }
+		inline const IDriverMemoryAllocation* getBoundMemory() const override { return nullptr; }
 		inline size_t getBoundMemoryOffset() const override { return 0ull; }
```

Comment:
do not couple, only OpenGL couples

Reply:
how does this even compile with the `override` keyword?

Reply:
because:
```C
class IGPUImage : public core::impl::ResolveAlignment<IDriverMemoryBacked,asset::IImage>, public IBackendObject
```
and these `virtual` belong to `IDriverMemoryBacked`